### PR TITLE
[FEATURE] Run profiler from DataContext

### PR DESCRIPTION
**Pull request by cdkini**
Changes proposed in this pull request:
- Run profiler method on DataContext
- Related tests
- Usage stats integration

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/rule_based_profiler/toolkit.py [hunk 1]
```diff
@@ -19,6 +20,42 @@
...
+            raise ge_exceptions.ProfilerError(
+                "Amibiguous arguments provided; you may either pass a RuleBasedProfiler or retrieve one from your ProfilerStore with a name/ge_cloud_id"
+            )
+    else:
+        profiler = get_profiler(
+            data_context=data_context,
+            profiler_store=profiler_store,
+            name=name,
+            ge_cloud_id=ge_cloud_id,
+        )
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/toolkit.py:48]: Remove `profiler` arg here - we just want to retrieve from store.

File: great_expectations/data_context/data_context.py [hunk 2]
```diff
@@ -3330,6 +3330,31 @@ def list_profilers(self) -> List[str]:
             ge_cloud_mode=self.ge_cloud_mode,
         )
 
+    @usage_statistics_enabled_method(
+        event_name="data_context.run_profiler",
+    )
+    def run_profiler(
```
Comment (cdkini) [inline great_expectations/data_context/data_context.py:3336]: Rename to `run_profiler_with_dynamic_arguments`

File: great_expectations/data_context/data_context.py [hunk 3]
```diff
@@ -3330,6 +3330,31 @@ def list_profilers(self) -> List[str]:
...
+        self,
+        profiler: Optional[RuleBasedProfiler] = None,
```
Comment (cdkini) [inline great_expectations/data_context/data_context.py:3338]: Remove!

**Review by alexsherstinsky — APPROVED** [review]
LGTM!

**Review by Shinnnyshinshin — APPROVED** [review]
love it :) thank you @cdkini

**Pull request merged**